components/homme/src/theta-l/prim_state_mod.F90

```diff
@@ -976,4 +982,64 @@ subroutine prim_diag_scalars(elem,hvcoord,tl,n,t_before_advance,nets,nete)
  
 
 end subroutine prim_diag_scalars
+
+!doing extrema with level for all elems
+subroutine findExtremaWithLevel(elem,res,field,operation,n0,nets,nete)
```

Comment:
This works for only w_i, so probably the function name should reflect that. Alternatively, generalize this routine by keeping the MINVAL line you removed at prim_state_mod.F90:237. If you keep it as is, then res should be intent(out) only, since its input value is unused.

Reply:
In longer run we may use this routine for u,v.